src/main/java/org/folio/rest/impl/SamlAPI.java

```diff
@@ -216,22 +218,22 @@ private void doPostSamlCallback(String body, RoutingContext routingContext, Map<
 
     final SessionStore sessionStore = new DummySessionStore(routingContext.vertx(), routingContext.session());
     final VertxWebContext webContext = new VertxWebContext(routingContext, sessionStore);
-    final String relayState = getRelayState(routingContext, body);
-
+    final Cookie relayStateCookie = routingContext.request().getCookie(RELAY_STATE);
+    String relayStateCookieValue = "null";
     URI relayStateUrl;
     try {
-      assert (relayState != null); // this avoids a Sonar warning later on
-      relayStateUrl = new URI(relayState);
+      relayStateCookieValue = relayStateCookie.getValue();
+      relayStateUrl = new URI(relayStateCookieValue);
     } catch (Exception e) {
       asyncResultHandler.handle(Future.succeededFuture(PostSamlCallbackResponse.respond400WithTextPlain(
-          "Invalid relay state url: " + relayState)));
+          "Invalid url in relayState cookie: " + relayStateCookieValue)));
       return;
     }
     URI originalUrl = relayStateUrl;
     URI stripesBaseUrl = UrlUtil.parseBaseUrl(originalUrl);
 
-    Cookie relayStateCookie = routingContext.getCookie(RELAY_STATE);
-    if (relayStateCookie == null || !relayState.contentEquals(relayStateCookie.getValue())) {
+    final String relayState = getRelayState(routingContext, body);
+    if (relayState == null || relayState.length() != 36 || !relayStateCookieValue.endsWith(relayState)) {
```

Comment:
I might put all these into method that returns a boolean named something like `isValidRelayState` or similar to make it more readable perhaps.